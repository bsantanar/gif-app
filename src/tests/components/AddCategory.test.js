import {shallow} from 'enzyme'
import { AddCategory } from '../../components/AddCategory'


describe('Tests on <AddCategory />', () => {
    
    //const setCategories = () => {}
    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories} />)

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    })

    test('should show correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should change textbox', () => {
        const input = wrapper.find('input')
        const value = 'Hola Mundo'
        input.simulate('change', {
            target: {
                value
            }
        })
    })

    test('should not submit form', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategories).not.toHaveBeenCalled()
    })
    
    test('should call setCategories and clean textbox', () => {
        
        const input = wrapper.find('input')
        const value = 'Hola Mundo'
        input.simulate('change', {
            target: {
                value
            }
        })
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategories).toHaveBeenCalled()
        expect(input.prop('value')).toBe('')
    })
    
    


})

