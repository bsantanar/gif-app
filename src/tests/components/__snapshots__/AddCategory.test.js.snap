// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory /> should show correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</div>
`;
